keanu-project/src/main/java/io/improbable/keanu/network/BayesianNetwork.java

```diff
@@ -57,6 +52,13 @@ public Vertex getVertexByLabel(VertexLabel label) {
         return vertices;
     }
 
+    /**
+     * @return A list of all vertices in the network.
+     */
+    public List<Vertex> getAllVertices() {
+        return Collections.unmodifiableList(vertices);
```

Comment:
What sort of method are you thinking of? And what would be the issue with getting a list of vertices like this?

Reply:
Agreed - in fact the only way you are using it at present is to "Get any vertex in the network" (WriteDot.java:58) - so is it actually needed?

Reply:
I need a way to traverse all the vertices in a network given a network. This can be done by having a vertex in a network, but my understanding is that currently we only have methods for retrieving latent and observed vertices, while it's possible that a network doesn't have either of these. 

What do you guys think is the best approach to achieving the above?

Reply:
I think that's a fair point.